libsolidity/formal/SMTLib2Interface.cpp

```diff
@@ -145,7 +151,8 @@ string SMTLib2Interface::checkSatAndGetValuesCommand(vector<Expression> const& _
 		for (size_t i = 0; i < _expressionsToEvaluate.size(); i++)
 		{
 			auto const& e = _expressionsToEvaluate.at(i);
-			// TODO they don't have to be ints...
+			solAssert(m_variables.count(e.name), "");
+			solAssert(m_variables[e.name] == SMTVariableType::Integer, "");
```

Comment:
Most of them will actually be functions (mapping int to int or int to bool).

Reply:
Note that this code is not tested by travis.

Reply:
Ah sorry, they are of course not functions.

Reply:
The code below maps them to `int`.